promql/engine.go

```diff
@@ -3313,8 +3320,45 @@ func setOffsetForAtModifier(evalTime int64, expr parser.Expr) {
 	})
 }
 
+func detectHistogramStatsDecoding(expr parser.Expr) {
+	var readHistogramStats bool
+	parser.Inspect(expr, func(node parser.Node, path []parser.Node) error {
```

Comment:
i think i would feel safer if we would only inspect the leaf nodes and then check the path if the last function up the tree was a histogram_sum or _count. That way its stateless and independent of the traversal.

Reply:
Updated, ptal again.

Reply:
lgtm